src/index_intf.ml

```diff
@@ -128,7 +128,7 @@ module type S = sig
   val mem : t -> key -> bool
   (** [mem t k] is [true] iff [k] is bound in [t]. *)
 
-  val replace : t -> key -> value -> unit
+  val replace : ?overcommit:bool -> t -> key -> value -> unit
   (** [replace t k v] binds [k] to [v] in [t], replacing any existing binding of
       [k]. *)
```

Comment:
`?overcommit` probably wants a mention in the doc-comment. I'd also prefer `?overcommit:unit`, but perhaps I'm the only one.

Reply:
I changed it back to bool option, to make it more consistent with irmin-pack arguments (especially here https://github.com/mirage/irmin/pull/1183/files#diff-7e9a5902736413eca81eab1399b2e2af88d7b0ed48dd90597a9873f0bcc42730R61 I think it does not look nice to mix unit and bool optional arguments).

Reply:
It makes sense to me to have a mix of the two when certain flags have a clear default and others do not, but as you prefer.